src/Psalm/CodeLocation.php

```diff
@@ -124,21 +133,38 @@ public function __construct(
         $this->previous_location = $previous_location;
         $this->text = $selected_text;
 
+        /** @psalm-suppress ImpureMethodCall Actually mutation-free just not marked */
```

Comment:
Maybe adding this to PHPParser stub would be more practical?

Reply:
Hmm, do we have a PHPParser stub?

Reply:
We do :) It's pretty light for now though:
https://github.com/vimeo/psalm/blob/870ee4ff933cff92f5d34abb7ec66d908198316f/stubs/phpparser.phpstub#L10

Reply:
Whoops missed that, can I submit a second PR for this later? Since it seems I would basically have to copy over all of php-parser's classes, otherwise psalm complains about child methods not being immutable.
Or maybe I could instead submit a PR to PHP-Parser directly.

Reply:
nikic didn't show a lot of enthusiasm when I tried to PR things there so stubbing is the best I think. I'm fine with pushing it later

Reply:
OK then, will submit a PR with immutable stubs later for this then.